stackit/internal/services/serverbackup/schedule/resource.go

```diff
@@ -0,0 +1,539 @@
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error deleting backup schedule", fmt.Sprintf("Calling API: %v", err))
+		return
+	}
+	tflog.Info(ctx, "Serverbackup schedule deleted")
+
+	// Disable backups service in case there are no backups and no backup schedules.
+	err = disableBackupsService(ctx, &model, r.client)
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error deleting schedule", fmt.Sprintf("Disabling server backup service after deleting schedule: %v", err))
+		return
+	}
+}
+
+// ImportState imports a resource into the Terraform state on success.
+// The expected format of the resource import identifier is: // project_id,server_id,schedule_id
+func (r *scheduleResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	idParts := strings.Split(req.ID, core.Separator)
+	if len(idParts) != 3 || idParts[0] == "" || idParts[1] == "" || idParts[2] == "" {
+		core.LogAndAddError(ctx, &resp.Diagnostics,
+			"Error importing credential",
+			fmt.Sprintf("Expected import identifier with format [project_id],[server_id],[backup_schedule_id], got %q", req.ID),
+		)
+		return
+	}
+
+	intId, err := strconv.ParseInt(idParts[2], 10, 64)
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics,
+			"Error importing credential",
+			fmt.Sprintf("Expected backup_schedule_id to be int64, got %q", idParts[2]),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("project_id"), idParts[0])...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("server_id"), idParts[1])...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("backup_schedule_id"), intId)...)
+	tflog.Info(ctx, "Serverbackup schedule state imported")
+}
+
+func mapResourceFields(ctx context.Context, schedule *serverbackup.BackupSchedule, model *scheduleResourceModel) error {
+	if schedule == nil {
+		return fmt.Errorf("response input is nil")
+	}
+	if model == nil {
+		return fmt.Errorf("model input is nil")
+	}
+	if schedule.Id == nil {
+		return fmt.Errorf("response id is nil")
+	}
+
+	model.BackupScheduleId = types.Int64PointerValue(schedule.Id)
+	idParts := []string{
+		model.ProjectId.ValueString(),
+		model.ServerId.ValueString(),
+		strconv.FormatInt(model.BackupScheduleId.ValueInt64(), 10),
+	}
+	model.ID = types.StringValue(
+		strings.Join(idParts, core.Separator),
+	)
+	model.Name = types.StringPointerValue(schedule.Name)
+	model.Rrule = types.StringPointerValue(schedule.Rrule)
+	model.Enabled = types.BoolPointerValue(schedule.Enabled)
+	if schedule.BackupProperties == nil {
+		model.BackupProperties = nil
+		return nil
+	}
+	ids, diags := types.ListValueFrom(ctx, types.StringType, schedule.BackupProperties.VolumeIds)
+	if diags.HasError() {
+		return fmt.Errorf("failed to map hosts: %w", core.DiagsToError(diags))
```

Comment:
In general we try to keep error messages short and avoid repeating patterns such as `failed to` or `error` in, because this is already clear from the start.

E.g. we use `Error creating schedule: process payload: map hosts: ...` instead of `Error creating schedule: failed to process payload: failed to map hosts: failed to ...`